### ISSUE TITLE
ReadTheDocs build broken

### ISSUE THREAD

**Issue by C4ptainCrunch** [ics-py/ics-py#314]
We have a webhook that build `main` on every push and it seems currently broken. Logs are available [over there](https://readthedocs.org/projects/icspy/builds/16073879/).

I made this issue mainly to remind myself to try to fix it (and maybe add a test in github actions too)

**Comment by N-Coder** [comment]
I can take care of this while also working on the Converter system docs as I wanted to update our ancient Sphinx anyways. Could you add me on RTD as developer/maintainer/whatever so that I can add tracking for a branch for trying this?

**Comment by C4ptainCrunch** [comment]
A bit late, but i added you just now

**Comment by C4ptainCrunch** [comment]
Update: the buils was failing because :
1. We were trying to build `master` and not `main`
2. We are using poetry and RTD was only searching for a requirements.txt

For 2., i had to use a [hack](https://github.com/readthedocs/readthedocs.org/issues/4912#issuecomment-1110189914) but it works.

Now, the problem lies further: https://readthedocs.org/projects/icspy/builds/17036215/
It seems that we have a problem with `napoleon` that is not compatible with our version of Python. This is most likely caused by us requiring `sphinx<3` where the current version is `4.5`. So i guess the next step is to upgrade Sphinx :)

**Comment by N-Coder** [comment]
Actually there is no need to ever touch poetry if you don't want to as a simple `pip install .` should always work with a reasonably recent pip and handle the build via poetry transparently in the background, see [here](https://python-poetry.org/docs/pyproject/#poetry-and-pep-517). I guess that the extra packages should also work with the usual pip syntax.

**Comment by N-Coder** [comment]
Yeah, packaging with setuptools already used to be confusing and now with flit, hatch, poetry, pipx, pipenv and whatnot it feels like we're doing javascript libraries :see_no_evil: 

And don't worry about the typos in the PRs, it's good to hear from you. 😊
